### PR TITLE
chore: Unify Link usage

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to unify the `Link` component usage by refactoring it to handle both external and client-side navigation.

## How does it work?

For client-side navigation, we continue to work with the framework standards, using Next's Link after checking the `href` attribute.

For external navigation, we use `Link` normally as an anchor element.

## How to test it?

Even after these changes, links should work as before.

## Checklist

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#12](https://github.com/vtex-sites/gatsby.store/pull/12))*

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description

### CODE REVIEW

File: src/components/ui/Link/Link.tsx [hunk 1]
```diff
@@ -1,36 +1,57 @@
...
+  ref: Ref<HTMLAnchorElement> | undefined
+) {
+  const defaultProps = {
+    'data-fs-link': true,
+    'data-fs-link-variant': variant,
+    'data-fs-link-inverse': inverse,
+    className: styles.fsLink,
+  }
+
+  if (isExternalLink(href)) {
```
Comment (tlgimenes) [inline src/components/ui/Link/Link.tsx:40]: I feel like we could benefit from `useMemo` in here:
```suggestion
  const external = useMemo(() => isExternalLink(href), [href])
  if (external) {
```

**Review by eduardoformiga — APPROVED** [review]
Good Job, @lucasfp13  ✨ 

Everything looks as expected on my tests.

File: src/components/ui/Link/Link.tsx [hunk 2]
```diff
@@ -1,36 +1,62 @@
...
+  ref: Ref<HTMLAnchorElement> | undefined
+) {
+  const isInternalLink = useMemo(
+    () => href[0] === '/' && href[1] !== '/',
+    [href]
+  )
+
+  if (isInternalLink) {
+    return (
+      <NextLink passHref href={href}>
```
Comment (hellofanny) [inline src/components/ui/Link/Link.tsx:32]: TIL! [passHref](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag)

**Review by hellofanny — APPROVED** [review]
Well Done 🚀 🚀 🚀 !! Everything seems to work fine 💯

**Review by filipewl — APPROVED** [review]

**Pull request merged**